pkg/cloudprovider/plugins/aws/aws_converters.go

```diff
@@ -137,6 +138,63 @@ func convertFromSecurityGroupPair(cloudGroups []*ec2.UserIdGroupPair, managedSGs
 	return cloudResourceIDs, desc
 }
 
+func convertIngressToIpPermission(rules []*cloudresource.CloudRule, cloudSGNameToObj map[string]*ec2.SecurityGroup) (
+	[]*ec2.IpPermission, error) {
+	ipPermissions := make([]*ec2.IpPermission, 0)
+	for _, obj := range rules {
+		rule := obj.Rule.(*cloudresource.IngressRule)
+		if rule == nil {
+			continue
+		}
+		description, err := utils.GenerateCloudDescription(obj.NpNamespacedName)
+		if err != nil {
+			return nil, fmt.Errorf("unable to generate rule description, err: %v", err)
```

Comment:
If any of the rules do not contain description, you fail the entire conversion?

Reply:
This is generating description for Nephe rules, which shouldn't fail at all. In an error case, we do want to fail the entire conversion or operation so the cloud won't be in partial state.